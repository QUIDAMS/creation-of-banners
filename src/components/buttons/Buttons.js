import React from 'react';
import {exportComponentAsPNG} from 'react-component-export-image';
import PropTypes from 'prop-types';

function copyAsJSON(banner) {
  navigator.clipboard.writeText(JSON.stringify(banner));
}

function copyAsJSX(htmlString) {
  navigator.clipboard.writeText(htmlString);
}

const Buttons = ({banner, previewRef, htmlString}) => {
  return(
    <div className='buttons'>
      <button className='buttons__copy' onClick={() => copyAsJSX(htmlString)}>
        Скопировать баннер в буфер обмена, как html
      </button>
      <button className='buttons__copy' onClick={() => exportComponentAsPNG(previewRef)}>
        Сохранить картинку в png
      </button>
      <button className='buttons__copy' onClick={() => copyAsJSON(banner)}>
        Скопировать баннер как JSON
      </button>
    </div>
  )
}

Buttons.propTypes = {
  previewRef: PropTypes.object,
  banner: PropTypes.object,
  htmlString: PropTypes.string,
}

export default Buttons;
