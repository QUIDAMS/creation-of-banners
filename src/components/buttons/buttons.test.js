import React from 'react';
import Buttons from './Buttons';
import {shallow} from 'enzyme';

describe('Testing Buttons', () => {
  const props = {
    banner: {
      image: '',
      line1: '',
      line2: '',
      line3: '',
      color: '',
      topColorGradient: '',
      bottomColorGradient: '',
      url: '',
    }
  }
  it('render correctly correctly', () => {
    const editor = shallow(<Buttons banner={props}/>);
    expect(editor).toMatchSnapshot();
  });
});
