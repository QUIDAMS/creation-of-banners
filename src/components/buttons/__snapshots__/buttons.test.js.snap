// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Buttons render correctly correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="buttons__copy"
    onClick={[Function]}
  >
    Скопировать баннер в буфер обмена, как html
  </button>
  <button
    className="buttons__copy"
    onClick={[Function]}
  >
    Сохранить картинку в png
  </button>
  <button
    className="buttons__copy"
    onClick={[Function]}
  >
    Скопировать баннер как JSON
  </button>
</div>
`;
